lib/rubocop/cop/naming/constant_name.rb

```diff
@@ -60,9 +60,14 @@ def allowed_assignment?(value)
 
         def allowed_method_call_on_rhs?(node)
           node && node.send_type? &&
-            (node.receiver.nil? || !node.receiver.literal?)
+            (node.receiver.nil? || !literal_receiver?(node))
         end
 
+        def_node_matcher :literal_receiver?, <<-PATTERN
```

Comment:
I tried to make this pattern include all logic in `#allowed_method_call_on_rhs?`, but got some weird non-matches on the new case that will require me to debug the node pattern a bit.